util/fuzzysearch.go

```diff
@@ -0,0 +1,129 @@
+package util
+
+import (
+	"strings"
+
+	iex "github.com/jonwho/go-iex"
+)
+
+// compute the edit-distance aka cost between the two strings
+func Levenshtein(s, t string) int {
+	sLen := len(s)
+	tLen := len(t)
+	cost := 0
+
+	if sLen == 0 {
+		return tLen
+	}
+	if tLen == 0 {
+		return sLen
+	}
+
+	matrix := make([][]int, sLen+1)
+	for i := range matrix {
+		matrix[i] = make([]int, tLen+1)
+	}
+
+	for i := 0; i <= sLen; i++ {
+		matrix[i][0] = i
+	}
+	for j := 0; j <= tLen; j++ {
+		matrix[0][j] = j
+	}
+
+	for i := 1; i <= sLen; i++ {
+		sRune := s[i-1]
+
+		for j := 1; j <= tLen; j++ {
+
+			tRune := t[j-1]
+
+			if sRune == tRune {
+				cost = 0
+			} else {
+				cost = 1
+			}
+
+			matrix[i][j] = Min3(matrix[i-1][j]+1, matrix[i][j-1]+1, matrix[i-1][j-1]+cost)
+		}
+	}
+
+	return matrix[sLen][tLen]
+}
+
+// computes Soundex value for the string
+func Soundex(s string) string {
+	m := map[byte]string{
+		'B': "1", 'P': "1", 'F': "1", 'V': "1",
+		'C': "2", 'S': "2", 'K': "2", 'G': "2", 'J': "2", 'Q': "2", 'X': "2", 'Z': "2",
+		'D': "3", 'T': "3",
+		'L': "4",
+		'M': "5", 'N': "5",
+		'R': "6",
+	}
+
+	s = strings.ToUpper(s)
+
+	r := string(s[0])
```

Comment:
do `r` and `p` represent anything? as in is it possible to use more descriptive variables